test_contextual_lenses/encoders.py

```diff
@@ -0,0 +1,157 @@
+"""Encoder functions."""
+"""Fixed and learnable transformations for embedding sequences."""
+
+
+import flax
+from flax import nn
+
+import jax
+from jax import lax
+import jax.nn
+import jax.numpy as jnp
+
+import numpy as np
+
+
+def one_hot_encoder(batch_inds, num_categories=21, **encoder_fn_kwargs):
+  """Applies one-hot encoding from jax.nn."""
+
+  one_hots = jax.nn.one_hot(batch_inds, num_classes=num_categories)
+  
+  return one_hots
+
+
+class CNN(nn.Module):
+  """A simple 1D CNN model."""
+
+  def apply(self, x, N_layers, N_features, N_kernel_sizes):
+    
+    x = jnp.expand_dims(x, axis=2)
+
+    for layer in range(N_layers):
+      features = N_features[layer]
+      kernel_size = (N_kernel_sizes[layer], 1)
+      x = nn.Conv(x, features=features, kernel_size=kernel_size)
+      x = nn.relu(x)
+    
+    x = x[:, :, 0, :]
```

Comment:
use jnp.squeeze(x, axis=2) instead to remove dimension (and ensure it is in fact length==1)